.github/workflows/apiBreakTest.yml

```diff
@@ -0,0 +1,216 @@
+name: Public Interface Breakage Detection
+
+on:
+  pull_request:
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  buildAndCheckAPIBreakage:
+    name: Build and Check API Breakage
+    runs-on: macos-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 #v4.1.1
+      with:
+        fetch-depth: 0
+
+
+    - name: Setup and Run Swift API Diff
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      run: |
+        # Define the list of exceptions to filter out
+        exceptions=(
+          'has been added as a new enum case$'
+          'is now with @_spi$'
+        )
+        
+        # Define the mandatory patterns to filter out
+        mandatory_patterns=(
+          '^/\*'
+          '^$'
+        )
+        
+        # Function to apply patterns with grep
+        apply_patterns() {
+          local input="$1"
+          local output="$input"
+        
+          # Apply mandatory patterns
+          for pattern in "${mandatory_patterns[@]}"; do
+            output=$(echo "$output" | grep -v "$pattern")
+          done
+        
+          # Apply exceptions
+          for exception in "${exceptions[@]}"; do
+            output=$(echo "$output" | grep -v "$exception")
+          done
+        
+          echo "$output"
+        }
+
+        echo "Swift version: $(swift --version)"
+        echo "Swift package manager version: $(swift package --version)"
+        swift package resolve
+
+        # Ensure we are in the correct directory
+        cd $GITHUB_WORKSPACE
+        
+        # Run swift-api-diff commands here directly
+        NEW_API_DIR=$(mktemp -d)
+        OLD_API_DIR=$(mktemp -d)
+        SDK_PATH=$(xcrun --show-sdk-path)
+
+        # Get all library module names
```

Comment:
I would add a comment about why not all the modules are listed, i.e. not being able to analyse `aws-crt-swift`.
Otherwise we will probably forget in a couple of months 😆